Microsoft.Toolkit.Uwp.UI.Controls/TabbedCommandBar/TabbedCommandBar.xaml

```diff
@@ -0,0 +1,297 @@
+﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+                    xmlns:controls="using:Microsoft.Toolkit.Uwp.UI.Controls">
+
+    <ResourceDictionary.MergedDictionaries>
+        <ResourceDictionary Source="ms-appx:///Microsoft.Toolkit.Uwp.UI.Controls/TabbedCommandBar/TabbedCommandBarItem.xaml" />
+    </ResourceDictionary.MergedDictionaries>
+
+    <DataTemplate x:Key="NormalTabTemplate">
+        <NavigationViewItem Content="{Binding Header}" />
+    </DataTemplate>
+    <DataTemplate x:Key="ContextualTabTemplate">
+        <NavigationViewItem Content="{Binding Header}">
```

Comment:
I'm thinking this is where the Visibility issue is, as the the CommandBarItem's visibility isn't tied to what's being displayed in the NavigationView.

I tested in XAML Studio that both the WUX & MUXC NavigationViews at least will respect a NavigationViewItem's Visibility property and hide it in the Top level navigation mode and that it'll respond to binding. You may get away with just a `Visibility="{Binding Visibility}"` here, but I'm not sure? Would there be other things like `IsEnabled` that need to be mapped too?

Only case I'm not sure what'll happen if the Contextual Tab is selected and then disappears... But maybe your selection changed event logic handles that?

Reply:
> Only case I'm not sure what'll happen if the Contextual Tab is selected and then disappears... But maybe your selection changed event logic handles that?

Oops, I hadn't thought about that when I switched to binding the ribbon content control directly to the SelectedItem. I think the old SelectionChanged logic could have been modified to handle it pretty easily (it just needs to check if the item's visibility is `Collapsed` and instead select the first ribbon item) @michael-hawker Do I need to revert back to using SelectionChanged, or is there a way to add that logic and still use a binding?

Reply:
I would try binding the Visibility here like I mentioned on line 13 first to see if that works? It may just allow the passthru to work?

Reply:
Yup, that works. It also can handle when the currently selected tab is set to hidden.